src/server.c

```diff
@@ -3677,13 +3681,15 @@ uint64_t getCommandFlags(client *c) {
 
 /* If this function gets called we already read a whole
  * command, arguments are in the client argv/argc fields.
- * processCommand() execute the command or prepare the
- * server for a bulk read from the client.
+ *
+ * validateCommand performs static validation of the command
  *
  * If C_OK is returned the client is still alive and valid and
  * other operations can be performed by the caller. Otherwise
  * if C_ERR is returned the client was destroyed (i.e. after QUIT). */
-int processCommand(client *c) {
+int validateCommand(client *c, int *valid) {
+    *valid = 0;
```

Comment:
i'm not sure it was necessary to extract these to a function, in #11012 we decided not to move it out and not to indent them all (just re-run some validations that may not be necessary).
but also, maybe some of these are wrong to skip, like the check about busy_module_yield_flags.

p.s. if we keep this change, "validate" may not be right, maybe "prepare" is better see e.g. moduleCallCommandFilters, and also, maybe the top comment still belongs to processCommand.

but anyway, i think after rebase we need to see if this change is still needed.

Reply:
maybe i don't understand the purpose of this function though.
i thought it's about command lookup, arity checks and other basic things that don't need to be re-evaluated when executing it the second time, but i now see that ACL checks are here too.

Reply:
now that we're rebased, why do we need this change?

Reply:
i rather revert it.
first to avoid unnecessary merge conflicts between branches and forks.
secondly, i think some of the code in this function in your current for is misplaced, like module command filters, or other things that may need to be re-processed.
lastly, if / when we do have to do some similar change due to some future project, the fine details of what needs to be in this function vs the outer one and it's name and role may be slightly different, so we'll have to modify things anyway, so why do it twice...